filter/fieldnamefinder_test.go

```diff
@@ -0,0 +1,25 @@
+package filter
+
+import (
+	"testing"
+
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFieldNameIter(t *testing.T) {
+	const typeString = "record[r1:record[r2:record[s:string,r3:record[t:time]],a:array[int64],r4:record[i:ip]]]"
+	typ, err := resolver.NewContext().LookupByName(typeString)
+	require.NoError(t, err)
+	var f fieldNameIter
+	f.init(typ.(*zng.TypeRecord))
+	require.Exactly(t, "r1.r2.s", string(f.next()))
+	require.False(t, f.done())
+	require.Exactly(t, "r1.r2.r3.t", string(f.next()))
+	require.False(t, f.done())
+	require.Exactly(t, "r1.a", string(f.next()))
+	require.False(t, f.done())
+	require.Exactly(t, "r1.r4.i", string(f.next()))
+	require.True(t, f.done())
+}
```

Comment:
Would you please add a `require` here that proves field names are matched case insensitively? (I see the code does this.)

Reply:
I can't really test that here, but the cases at https://github.com/brimsec/zq/pull/1188/files#diff-0d6694b2e41de6ea801cf99cbb136243R462-R473 do test it.